src/redisearch_rs/trie_bencher/build.rs

```diff
@@ -5,20 +5,26 @@ fn main() {
     let root = git_root();
 
     // Construct the correct folder path based on OS and architecture
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
     let lib_dir = {
         let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
         let target_arch = match target_arch.as_str() {
             "aarch64" => "arm64v8",
             "x86_64" => "x64",
             _ => &target_arch,
         };
-        let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+
         root.join(format!(
             "bin/{target_os}-{target_arch}-release/search-community/deps/triemap"
         ))
     };
 
     assert!(std::fs::exists(lib_dir.join("libtrie.a")).unwrap());
+    // Allow undefined symbols in non-macOS builds to align with macOS linker behavior, which permits unresolved symbols by default.
+    // This is safe here because the undefined symbols are resolved at runtime via dynamic linking, ensuring compatibility in CI on Ubuntu.
```

Comment:
```suggestion
    // There are several symbols exposed by `libtrie.a` that we don't
    // actually invoke (either directly or indirectly) in our benchmarks.
    // We provide a definition for the ones we need (e.g. Redis' allocation functions),
    // but we don't want to be forced to add dummy definitions for the ones we don't rely on.
    // We prefer to fail at runtime if we try to use a symbol that's undefined.
    // This is the default linker behaviour on macOS. On other platforms, the default
    // configuration is stricter: it exits with an error if any symbol is undefined. 
    // We intentionally relax it here.
```

Let's explain the _why_ in comments, not just the how.